tests/end_to_end_tests/verify.py

```diff
@@ -0,0 +1,309 @@
+
+        return query
+
+    def send(self):
+        query = self.build()
+
+        protocol = "https://" if not self._server_address.startswith("http") else ""
+
+        full_query = "{0}{1}/api/query?queryType=log&{2}".format(
+            protocol, self._server_address, query
+        )
+
+        self._logger.info("Query server: {0}".format(full_query))
+
+        with requests.Session() as session:
+            resp = session.get(full_query)
+
+        if resp.status_code != 200:
+            self._logger.info(f"Query failed with {resp.text}.")
+            return None
+
+        data = resp.json()
+
+        return data
+
+
+TEST_LOG_MESSAGE_COUNT = 1000
+
+
+_QUERY_RETRY_DELAY = 10
+
+
+def verify_logs(
+    scalyr_api_read_key: str,
+    scalyr_server: str,
+    get_agent_log_content: Callable[[], str],
+    counters_verification_query_filters: List[str],
+    counter_getter: Callable[[Any], int],
+    write_counter_messages: Callable[[], None] = None,
+    verify_ssl: bool = True,
+):
+    """
+    Do a basic verifications on agent log file.
+    It also writes test log with counter messages that are queried later from Scalyr servers to compare results.
+    :param scalyr_api_read_key: Scalyr API key with read permissions.
+    :param scalyr_server: Scalyr server hostname.
+    :param get_agent_log_content: Function that has to return current contant of the running agent log.
+        That function has to implemented accoring to a type of the running agent, e.g. kubernetes, docker, or package
+    :param counters_verification_query_filters:  List of Scalyr query language filters which are required to fetch
+        messages that are ingested by the 'write_counter_messages'
+    :param counter_getter: Function which should return counter from the ingested message.
+    :param write_counter_messages: Function that writes counter messages to upload the to Scalyr.
+        Can be None, for example for the kubernetes image test, where writer pod is already started.
+    """
+    if write_counter_messages:
+        log.info("Write test log file messages.")
+        write_counter_messages()
+
+    agent_log_content = get_agent_log_content()
+
+    if verify_ssl:
+        # Verify agent start up line
+        assert "Starting scalyr agent" in agent_log_content
+        # Ensure CA validation is not disabled with default install
+        assert "sslverifyoff" not in agent_log_content
+        assert (
+            "Server certificate validation has been disabled" not in agent_log_content
+        )
+
+    check_agent_log_for_errors(content=agent_log_content)
+
+    log.info("Wait for agent log requests stats.")
+    while not check_requests_stats_in_agent_log(content=get_agent_log_content()):
+        time.sleep(1)
+
+    log.info(
+        "Verify that previously written test log file content has been uploaded to server."
+    )
+    while True:
+        resp = ScalyrQueryRequest(
```

Comment:
Same here, I also don't see any timeout for this while loop, but maybe I missed it.

Reply:
https://github.com/scalyr/scalyr-agent-2/pull/951#discussion_r935285122